### ISSUE TITLE
HTML parser: certain nested HTML elements cause fatal crash

### ISSUE THREAD

**Issue by a-murph** [remarkjs/react-markdown#302]
Nesting certain HTML elements within the source string that is sent to ReactMarkdown causes a fatal crash. In particular I have noticed issues with the mark and span tags, though even divs can cause this crash in some circumstances.

This occurs when passing a string that includes HTML tags as the `source` prop to the ReactMarkdown element, with `escapeHtml` set to `false`. The project in which I encountered this issue is using ReactMarkdown 4.0.6, though the same behavior is replicable in the [live editor](https://rexxars.github.io/react-markdown/), replacing the example blockquote with other HTML.

Here are some examples of HTML that causes ReactMarkdown to crash:
- `<span><span>Text</span></span>`
- `<div><div>Text</div></div>`
- `<mark><mark>Text</mark></mark>`

divs seem to usually avoid the crash if there is whitespace between the tags. The following all render properly:
- `<div> <div> Text </div> </div>`
- `<div> <div>Text</div> </div>`
- `<div><div>Text</div> </div>`

Although this crashes:
- `<div><div> Text </div></div>`

spans, marks, and other inline elements crash even with spacing. The following all crash:
- `<mark> <mark> Text </mark> </mark>`
- `<span> <span> Text </span> </span>`
- `<em> <em> Text </em> </em>`

Here is the error that my project produces in the console when this crash occurs:

> ast-to-react.js:214 Uncaught TypeError: Cannot read property 'props' of undefined
>     at mergeNodeChildren (ast-to-react.js:214)
>     at getNodeProps (ast-to-react.js:175)
>     at astToReact (ast-to-react.js:18)
>     at ast-to-react.js:23
>     at Array.map (<anonymous>)
>     at resolveChildren (ast-to-react.js:22)
>     at astToReact (ast-to-react.js:19)
>     at ast-to-react.js:23
>     at Array.map (<anonymous>)
>     at resolveChildren (ast-to-react.js:22)

**Comment by Muwbi** [comment]
+1

**Comment by polskikiel** [comment]
In my case two `<img>` tags in one line causes the crash of the ReactMarkdown. On the live editor I get the following output: 
```
TypeError: Cannot read property 'start' of undefined
    at demo.js:14
    at demo.js:14
    at demo.js:1
    at a (demo.js:14)
    at demo.js:14
    at a (demo.js:14)
    at demo.js:14
    at a (demo.js:14)
    at s (demo.js:14)
    at a (demo.js:1)
```
If I make a newline between the tags the ReactMarkdown handle that.

**Comment by ianhe8x** [comment]
it was claimed solved in #43 but unfortunately not.

**Comment by jorgeorpinel** [comment]
We're having a similar problem with `<abbr>` elements inside `<details>`. See iterative/dvc.org/issues/510 for context. Hoping for a resolution 🤞

**Comment by ChristianMurphy** [comment]
If merged #428 would allow this to be fixed.

**Comment by wooorm** [comment]
This should be solved by landing GH-563 today, which will soon be released in v6.0.0!
